src/resource.js

```diff
@@ -1,6 +1,58 @@
 'use strict'
 
-export default () => {
+import url from 'url'
+import jts from 'jsontableschema'
+import _ from 'lodash'
 
-}
+export default class Resource {
 
+  constructor(descriptor) {
+    this._descriptor = descriptor
+    this.REMOTE_PROTOCOLS = ['http:', 'https:', 'ftp:', 'ftps:']
+  }
+
+  getDescriptor() {
+    return this._descriptor
+  }
+
+  getName() {
+    return this.getDescriptor()['name']
+  }
+
+  getType() {
+    const resourceField = this._getResourceField()
+
+    if (resourceField === 'data') {
+      return 'inline'
+    }
+
+    const protocol = url.parse(this.getDescriptor()[resourceField])['protocol']
+    if (_.includes(this.REMOTE_PROTOCOLS, protocol)) {
+      return 'remote'
+    } else {
+      return 'local'
+    }
+
+  }
+
+  getSource() {
+    return this.getDescriptor()[this._getResourceField()]
+  }
+
+  getTable() {
```

Comment:
Well I have overlooked the fact that `await` can be only used with `async` functions which return `AsyncFunction` object which is basically also a promise.

So for now I will use `await` in our code and let resource.table just return the promise.